protocol/native/native.go

```diff
@@ -36,7 +38,28 @@ func ExecParser(toNoder ToNoder, path string, args ...string) (*Parser, error) {
 }
 
 func (p *Parser) ParseUAST(req *protocol.ParseUASTRequest) *protocol.ParseUASTResponse {
-	nativeResp, err := p.Client.ParseNativeAST(&ParseASTRequest{Content: req.Content})
+	// If the request encoding is Base64 try to decode it as UTF8. Leave it
+	// as Base64 if it isn't possible.
```

Comment:
Sorry, I don't understand what you mean. As I understand it, only the native parser may be able to handle the file encoding (i. e. for a binary source code), and that's ok. I don't know if you mean that this assumption is not correct or anything else.